File: c++/119.cpp
#include <iostream>
#include <string>
#include <map>
#include <sstream>
#include <vector>

using namespace std;

vector<string> names;
vector<string> guy;
map<string, int> gifts;

void split(string &line, vector<string> &v) {
    v.clear();
    stringstream ss(line);
    string x;
    while(ss >> x) v.push_back(x);
}

int main() {
    int T;
    string line;
    while(cin >> T) {
        getline(cin, line);
        split(line, names);
        for(int i = 0; i < T; i++) {

        }
    }
}

File: python/119.py
while True:
	try:
		T = int(input())
	except EOFError:
		break
	names = input().split()
	gifts = {a : 0 for a in names}
	for _ in range(T):
		guy = input().split()
		name = guy[0]
		give = int(guy[1])
		if give == 0:
			continue
		del guy[0]
		del guy[0]
		gifts[name] -= give
		total = int(guy[0])
		del guy[0]
		gives_each = give // total
		gifts[name] += give - gives_each * total
		for person in guy:
			gifts[person] += gives_each
	for name in gifts:
		print(name, gifts[name])
	print()